app/src/main/java/org/simple/clinic/navigation/v2/Router.kt

```diff
@@ -270,12 +270,11 @@ class Router(
       fragment: Fragment,
       lastButOneScreen: NavRequest?
   ) {
-    val isNeitherTopNorLastButOneScreen = navRequest != newTopScreen && navRequest != lastButOneScreen
```

Comment:
Yep, good point. we can reintroduce this now that we are using last full screen instead of last but one screen. That should ensure we are not removing modals and only removing full screens (since we don't use modals for a deep navigations paths it should be fine to retain them). wdyt?

Reply:
NVM, this will remove those modals as well.